src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -117,6 +118,22 @@ public void SetProperty<T>(string propertyName, T value)
             Properties[propertyName] = new SerializedPropertyInfo(serializedValue, isString);
         }
 
+        public void SetPropertiesFrom(IPropertyBagHolder other)
```

Comment:
Oh right, this is the reason for this whole PR. Adding this API to ensure properties of all types (not just strings) can be copied from one holder to another, as required by the RuleDictionaryConverter.